serializable.ts

```diff
@@ -190,9 +197,16 @@ function fromJSON<T>(
       continue;
     }
 
-    accumulator[propertyKey as keyof T] = Array.isArray(value)
-      ? value.map((v) => fromJSONStrategy(v))
-      : fromJSONStrategy(value as JSONValue);
+    // if strategy is not provided we apply the default on the value/elements of the array
+    if (!fromJSONStrategy) {
+      const defaultFromJSONStrategy = fromJSONDefault;
+      accumulator[propertyKey as keyof T] = Array.isArray(value)
+      ? value.map((v) => defaultFromJSONStrategy(v)) : defaultFromJSONStrategy(value as JSONValue)
+      continue;
+    }
+
+    // if a strategy is provided we apply the strategy on the entire array
+    accumulator[propertyKey as keyof T] = fromJSONStrategy(value as JSONValue);
```

Comment:
this is basically it now because we want to pass the direct value through the strat array or not, and we and guarantee the default set by the destructuring.